src/main/scala/chiseltest/stage/phases/CompileDut.scala

```diff
@@ -0,0 +1,27 @@
+package chiseltest.stage.phases
+
+import chiseltest.backends.{TreadleBackend, VcsBackend, VerilatorBackend}
+import chiseltest.stage.ChiselTestOptions
+import firrtl.AnnotationSeq
+import firrtl.options.{Dependency, Phase, Viewer}
+
+/** add [[chiseltest.backends.SimulatorInterfaceAnnotation]] to annotations. */
+class CompileDut extends Phase {
+  override def prerequisites: Seq[Dependency[Phase]] = Seq(Dependency[AnalysisCircuit])
+
+  override def optionalPrerequisites = Seq.empty
+
+  override def optionalPrerequisiteOf = Seq.empty
+
+  override def invalidates(a: Phase) = false
+
+  override def transform(annotations: AnnotationSeq): AnnotationSeq = {
+    val options = Viewer[ChiselTestOptions].view(annotations)
+    options.backend.get match {
+      case "treadle"   => TreadleBackend.compileDut(annotations)
+      case "verilator" => VerilatorBackend.compileDut(annotations)
+      case "vcs"       => VcsBackend.compileDut(annotations)
+    }
```

Comment:
It would be nice if there was a more typesafe way of doing this. Not critical to fix now, though.